bin/sunnyportal2file

```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Joel Berglund <joebe975@protonmail.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+# USA
+
+from datetime import date, datetime, timedelta
+from getpass import getpass
+
+import argparse
+import configparser
+import logging
+import sunnyportal.client, sunnyportal.responses
+import re
+import os
+import pandas as pd
+import numpy as np
+
+def valid_date_type(arg_date_str):
+    try:
+        m = re.match("(?P<year>\d{4})-(?P<month>\d{1,2})-(?P<day>\d{1,2})", arg_date_str)
+        d = date(int(m.group('year')),int(m.group('month')),int(m.group('day')))
```

Comment:
Can you use `datetime.strptime` instead?

Reply:
Resolved by 7bec61c.